feat: `bench setup requirements --dev`

ankush opened the pull request:
Often applications have development or test specific requirements which are not required in production.

Changes:
- Add new flag `bench setup requirements --dev`
- installs all `dev-requirements.txt` in root folder of apps.
- While doing `bench get-app` if dev mode is enabled in common_site_config then dev requirements are also installed. 


Documentation: https://github.com/frappe/frappe_docs/pull/196

ankush committed:
chore: typo

ankush committed:
chore: update project description

ankush committed:
feat: command to install dev-requirements.txt

Often applications have development or test specific requirements which
are not required in production.

- Add new command `bench setup dev-requirements`
- installs all `dev-requirements.txt` in app's root folder.

gavindsouza commented on bench/utils.py:
```diff
@@ -617,6 +617,26 @@ def update_node_packages(bench_path='.'):
+
+
+def _get_venv_python(bench_path="."):
+	return os.path.join(bench_path, "env", "bin", "python")
```
There's `bench.utils.get_env_cmd("python")` for this

gavindsouza commented on bench/utils.py:
```diff
@@ -617,6 +617,26 @@ def update_node_packages(bench_path='.'):
+			print(f'\n{color.yellow}Installing development dependencies for {app}{color.nc}')
+			exec_cmd(f"{env_py} -m pip install -q -r {dev_requirements_path}", cwd=bench_path)
+		else:
+			print(f'\n{color.red} dev-requirements.txt not found in {app}{color.nc}')
```
Use `bench.utils.log` instead. Common interface to log bench commands etc and add colour.

gavindsouza commented on bench/commands/setup.py:
```diff
@@ -148,6 +148,11 @@ def setup_requirements(node=False, python=False):
 		from bench.utils import update_node_packages
 		update_node_packages()
 
+@click.command("dev-requirements", help="Setup optional development dependencies")
```
I think we don't need another command for this. This can be a part of the `setup-requirements` command.

Eg: `bench setup requirements --dev` to setup dev + prod requirements it for all bench apps; fulfilling Python and Node dependencies. `bench setup requirements --dev --python` for only the Python dependencies, etc.

ankush replied:
Not sure if we can handle `node`, as dev requirements are installed right now and changing that behaviour could be a breaking change. 😐 

Unlike python req.txt where we are proposing a new separate file, node deps are all stored in `package.json`.

gavindsouza replied:
We could probably though...not for Frappe, but for other apps. Else, we could just put a yellow line/comment saying `--dev` doesn't support fulfilling Node dev deps yet?

ankush replied:
Something along lines of "node dev deps are installed by default"

gavindsouza reviewed (commented):
This reminded me of back when I tried reducing node_modules size by ignoring the dev requirements in prod. Maybe in the future 🥲

https://github.com/frappe/bench/pull/922

ankush committed:
refactor: remove duplicate function

ankush committed:
refactor: use `log` instead of print

ankush committed:
refactor: merge dev-requirement command

ankush committed:
feat: install dev-dependencies in get-app if dev

When developer mode is enabled install all dev dependencies too while
doing `get-app`

ankush committed:
fix: warn about --dev not supporting node
